### PR TITLE
[api-migration-hackathon] MandatoryCriticalPoints migration

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR migrates the MandatoryCriticalPoints module:

- the new log API is used,
- the new architecture is in place,
- arrays are now set with `Set|GetInputArrayToProcess`,
- the standalone has also been migrated.

Due to compatibility issues, two statements in the ContourTree module have been temporary commented (AbstractTriangulation is always empty…).

The LowestCommonAncestor and RangeMinimumQuery module have also been updated.

Corresponding ttk-data PR: https://github.com/topology-tool-kit/ttk-data/pull/48

Enjoy,
Pierre

### CODE REVIEW

File: core/vtk/ttkMandatoryCriticalPoints/vtk.module [hunk 1]
```diff
@@ -1,6 +1,4 @@
 NAME
  ttkMandatoryCriticalPoints
 DEPENDS
-  VTK::FiltersCore
-PRIVATE_DEPENDS
-  VTK::CommonCore
+ ttkAlgorithm
```
Comment (vollia) [inline core/vtk/ttkMandatoryCriticalPoints/vtk.module:4]: @julien-tierny this change is what I am mentioning in #451 
@pierre-guillou do not worry, if this PR is merged I can rebase mine and fix this on my side

Comment (vollia) [reply core/vtk/ttkMandatoryCriticalPoints/vtk.module:4]: Never-mind, we can keep the mechanism with `ttkAlgorithm`.

**Comment by julien-tierny** [comment]
awesome, thanks!

**Comment by julien-tierny** [comment]
@pierre-guillou 
could you also migrate the logging for core/base/contourTree as well? 
(the legacy one is still polluting the console output).

**Comment by julien-tierny** [comment]
plus there seems to be some double-logging in core/base/lowestCommonAncestor

**Comment by pierre-guillou** [comment]
> @pierre-guillou
> could you also migrate the logging for core/base/contourTree as well?
> (the legacy one is still polluting the console output).

Will do. I thought it was a dependency of FTM and/or ContourForests and that Charles or Jonas would deal with this module.

> plus there seems to be some double-logging in core/base/lowestCommonAncestor

There is one LowestCommonAncestor method that is called twice by MandatoryCriticalPoints (preprocess). I reduced the warning level in this method so it won't appear by default.

**Pull request merged**